[BUG] cmake build fails at src/timeout.c:169 (`-Werror,-Wimplicit-const-int-float-conversion`)

While attempting to build latest unstable branch with cmake, it fails with:

```bash
/Users/jbergstroem/wrk/oss/valkey/src/timeout.c:169:21: error: implicit conversion from 'long long' to 'long double' changes value from 9223372036854775807 to 9223372036854775808 [-Werror,-Wimplicit-const-int-float-conversion]
  169 |         if (ftval > LLONG_MAX) {
      |                   ~ ^~~~~~~~~
/opt/homebrew/Cellar/llvm/19.1.7_1/lib/clang/19/include/limits.h:109:20: note: expanded from macro 'LLONG_MAX'
  109 | #define LLONG_MAX  __LONG_LONG_MAX__
      |                    ^~~~~~~~~~~~~~~~~
<built-in>:63:27: note: expanded from macro '__LONG_LONG_MAX__'
   63 | #define __LONG_LONG_MAX__ 9223372036854775807LL
      |                           ^~~~~~~~~~~~~~~~~~~~~
1 error generated.
```

**To reproduce**

```bash
git checkout 3f6581b62 # latest unstable at point of testing
mkdir build && cd build
cmake .. -DCMAKE_BUILD_TYPE=Debug # build type doesn't matter
make
```


Any additional information that is relevant to the problem.

- MacOS 15.3.1 (24D70)
- clang version 19.1.7 installed via homebrew (same issue on llvm@18)
- cmake 3.31.6

--- 

Full build log:

```bash
~/w/o/valkey/build unstable $ git rev-parse --short HEAD
3f6581b62
~/w/o/valkey/build unstable $ cmake .. -DCMAKE_BUILD_TYPE=Debug
-- Found /opt/homebrew/opt/llvm/bin/clang++, /opt/homebrew/opt/llvm/bin/clang
-- The C compiler identification is Clang 19.1.7
-- The CXX compiler identification is Clang 19.1.7
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /opt/homebrew/opt/llvm/bin/clang - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /opt/homebrew/opt/llvm/bin/clang++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Processor count: 12
-- Building in debug mode
-- Using libc
-- TLS is disabled
-- RDMA is disabled
-- Building on Darwin
-- Compiling valkey for ARM64
-- Looking for include file stdatomic.h
-- Looking for include file stdatomic.h - found
-- Looking for mkstemp
-- Looking for mkstemp - found
CMake Deprecation Warning at deps/hiredis/CMakeLists.txt:1 (CMAKE_MINIMUM_REQUIRED):
  Compatibility with CMake < 3.10 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.


Detected version: 1.2.0
-- Found python3: /opt/homebrew/bin/python3
-- Adding target generate_commands_def
-- Adding target generate_fmtargs_h
-- Adding target generate_test_files_h
-- CFLAGS:   -fno-omit-frame-pointer -std=gnu11 -pedantic
-- Valkey version: 255.255.255
-- Current host distro: darwin
-- Configuring done (0.9s)
-- Generating done (0.1s)
-- Build files have been written to: /Users/jbergstroem/wrk/oss/valkey/build
~/w/o/valkey/build unstable $ make
[  1%] Building C object deps/lua/CMakeFiles/lualib.dir/src/fpconv.c.o
[  1%] Building C object deps/lua/CMakeFiles/lualib.dir/src/lbaselib.c.o
[  2%] Building C object deps/lua/CMakeFiles/lualib.dir/src/lmathlib.c.o
[  2%] Building C object deps/lua/CMakeFiles/lualib.dir/src/lstring.c.o
[  3%] Building C object deps/lua/CMakeFiles/lualib.dir/src/lparser.c.o
[  3%] Building C object deps/lua/CMakeFiles/lualib.dir/src/ldo.c.o
[  4%] Building C object deps/lua/CMakeFiles/lualib.dir/src/lzio.c.o
[  4%] Building C object deps/lua/CMakeFiles/lualib.dir/src/lmem.c.o
[  5%] Building C object deps/lua/CMakeFiles/lualib.dir/src/strbuf.c.o
[  5%] Building C object deps/lua/CMakeFiles/lualib.dir/src/lstrlib.c.o
[  6%] Building C object deps/lua/CMakeFiles/lualib.dir/src/lundump.c.o
[  6%] Building C object deps/lua/CMakeFiles/lualib.dir/src/lua_cmsgpack.c.o
[  7%] Building C object deps/lua/CMakeFiles/lualib.dir/src/loslib.c.o
[  7%] Building C object deps/lua/CMakeFiles/lualib.dir/src/lua_struct.c.o
[  8%] Building C object deps/lua/CMakeFiles/lualib.dir/src/ldebug.c.o
[  8%] Building C object deps/lua/CMakeFiles/lualib.dir/src/lobject.c.o
[  9%] Building C object deps/lua/CMakeFiles/lualib.dir/src/ldump.c.o
[  9%] Building C object deps/lua/CMakeFiles/lualib.dir/src/lua_cjson.c.o
[ 10%] Building C object deps/lua/CMakeFiles/lualib.dir/src/ldblib.c.o
[ 10%] Building C object deps/lua/CMakeFiles/lualib.dir/src/ltm.c.o
[ 11%] Building C object deps/lua/CMakeFiles/lualib.dir/src/ltable.c.o
[ 11%] Building C object deps/lua/CMakeFiles/lualib.dir/src/lstate.c.o
[ 12%] Building C object deps/lua/CMakeFiles/lualib.dir/src/lua_bit.c.o
[ 12%] Building C object deps/lua/CMakeFiles/lualib.dir/src/lua.c.o
[ 13%] Building C object deps/lua/CMakeFiles/lualib.dir/src/loadlib.c.o
[ 13%] Building C object deps/lua/CMakeFiles/lualib.dir/src/lcode.c.o
[ 14%] Building C object deps/lua/CMakeFiles/lualib.dir/src/lapi.c.o
[ 14%] Building C object deps/lua/CMakeFiles/lualib.dir/src/lgc.c.o
[ 15%] Building C object deps/lua/CMakeFiles/lualib.dir/src/lvm.c.o
[ 15%] Building C object deps/lua/CMakeFiles/lualib.dir/src/lfunc.c.o
[ 16%] Building C object deps/lua/CMakeFiles/lualib.dir/src/lauxlib.c.o
[ 16%] Building C object deps/lua/CMakeFiles/lualib.dir/src/ltablib.c.o
[ 17%] Building C object deps/lua/CMakeFiles/lualib.dir/src/linit.c.o
[ 17%] Building C object deps/lua/CMakeFiles/lualib.dir/src/lopcodes.c.o
[ 18%] Building C object deps/lua/CMakeFiles/lualib.dir/src/llex.c.o
[ 18%] Building C object deps/lua/CMakeFiles/lualib.dir/src/liolib.c.o
[ 19%] Linking C static library ../../lib/liblualib.a
[ 19%] Built target lualib
[ 20%] Building C object deps/hiredis/CMakeFiles/hiredis.dir/alloc.c.o
[ 20%] Building C object deps/hiredis/CMakeFiles/hiredis.dir/async.c.o
[ 21%] Building C object deps/hiredis/CMakeFiles/hiredis.dir/hiredis.c.o
[ 21%] Building C object deps/hiredis/CMakeFiles/hiredis.dir/net.c.o
[ 22%] Building C object deps/hiredis/CMakeFiles/hiredis.dir/read.c.o
[ 22%] Building C object deps/hiredis/CMakeFiles/hiredis.dir/sds.c.o
[ 23%] Building C object deps/hiredis/CMakeFiles/hiredis.dir/sockcompat.c.o
[ 23%] Linking C static library ../../lib/libhiredisd.a
[ 23%] Built target hiredis
[ 24%] Building C object deps/linenoise/CMakeFiles/linenoise.dir/linenoise.c.o
[ 24%] Linking C static library ../../lib/liblinenoise.a
[ 24%] Built target linenoise
[ 24%] Building C object deps/fpconv/CMakeFiles/fpconv.dir/fpconv_dtoa.c.o
[ 25%] Linking C static library ../../lib/libfpconv.a
[ 25%] Built target fpconv
[ 26%] Building C object deps/hdr_histogram/CMakeFiles/hdr_histogram.dir/hdr_histogram.c.o
[ 26%] Linking C static library ../../lib/libhdr_histogram.a
[ 26%] Built target hdr_histogram
[ 27%] Built target generate_fmtargs_h
#define REDIS_GIT_SHA1 "3f6581b6"
#define REDIS_GIT_DIRTY "       0"
[ 27%] Built target release_header
[ 27%] Built target generate_commands_def
[ 28%] Building C object src/CMakeFiles/valkey-server.dir/threads_mngr.c.o
[ 28%] Building C object src/CMakeFiles/valkey-server.dir/adlist.c.o
[ 29%] Building C object src/CMakeFiles/valkey-server.dir/quicklist.c.o
[ 29%] Building C object src/CMakeFiles/valkey-server.dir/ae.c.o
[ 30%] Building C object src/CMakeFiles/valkey-server.dir/anet.c.o
[ 30%] Building C object src/CMakeFiles/valkey-server.dir/dict.c.o
[ 31%] Building C object src/CMakeFiles/valkey-server.dir/hashtable.c.o
[ 32%] Building C object src/CMakeFiles/valkey-server.dir/kvstore.c.o
[ 32%] Building C object src/CMakeFiles/valkey-server.dir/sds.c.o
[ 33%] Building C object src/CMakeFiles/valkey-server.dir/zmalloc.c.o
[ 33%] Building C object src/CMakeFiles/valkey-server.dir/lzf_c.c.o
[ 34%] Building C object src/CMakeFiles/valkey-server.dir/lzf_d.c.o
[ 34%] Building C object src/CMakeFiles/valkey-server.dir/pqsort.c.o
[ 35%] Building C object src/CMakeFiles/valkey-server.dir/zipmap.c.o
[ 35%] Building C object src/CMakeFiles/valkey-server.dir/sha1.c.o
[ 36%] Building C object src/CMakeFiles/valkey-server.dir/ziplist.c.o
[ 36%] Building C object src/CMakeFiles/valkey-server.dir/release.c.o
[ 37%] Building C object src/CMakeFiles/valkey-server.dir/memory_prefetch.c.o
[ 37%] Building C object src/CMakeFiles/valkey-server.dir/io_threads.c.o
[ 38%] Building C object src/CMakeFiles/valkey-server.dir/networking.c.o
[ 38%] Building C object src/CMakeFiles/valkey-server.dir/util.c.o
[ 39%] Building C object src/CMakeFiles/valkey-server.dir/object.c.o
[ 39%] Building C object src/CMakeFiles/valkey-server.dir/db.c.o
[ 40%] Building C object src/CMakeFiles/valkey-server.dir/replication.c.o
[ 40%] Building C object src/CMakeFiles/valkey-server.dir/rdb.c.o
[ 41%] Building C object src/CMakeFiles/valkey-server.dir/t_string.c.o
[ 41%] Building C object src/CMakeFiles/valkey-server.dir/t_list.c.o
[ 42%] Building C object src/CMakeFiles/valkey-server.dir/t_set.c.o
[ 42%] Building C object src/CMakeFiles/valkey-server.dir/t_zset.c.o
[ 43%] Building C object src/CMakeFiles/valkey-server.dir/t_hash.c.o
[ 43%] Building C object src/CMakeFiles/valkey-server.dir/config.c.o
[ 44%] Building C object src/CMakeFiles/valkey-server.dir/aof.c.o
[ 44%] Building C object src/CMakeFiles/valkey-server.dir/pubsub.c.o
[ 45%] Building C object src/CMakeFiles/valkey-server.dir/multi.c.o
[ 45%] Building C object src/CMakeFiles/valkey-server.dir/debug.c.o
[ 46%] Building C object src/CMakeFiles/valkey-server.dir/sort.c.o
[ 46%] Building C object src/CMakeFiles/valkey-server.dir/intset.c.o
[ 47%] Building C object src/CMakeFiles/valkey-server.dir/syncio.c.o
[ 47%] Building C object src/CMakeFiles/valkey-server.dir/cluster.c.o
[ 48%] Building C object src/CMakeFiles/valkey-server.dir/cluster_legacy.c.o
[ 48%] Building C object src/CMakeFiles/valkey-server.dir/cluster_slot_stats.c.o
[ 49%] Building C object src/CMakeFiles/valkey-server.dir/crc16.c.o
[ 49%] Building C object src/CMakeFiles/valkey-server.dir/endianconv.c.o
[ 50%] Building C object src/CMakeFiles/valkey-server.dir/commandlog.c.o
[ 50%] Building C object src/CMakeFiles/valkey-server.dir/eval.c.o
[ 51%] Building C object src/CMakeFiles/valkey-server.dir/bio.c.o
[ 51%] Building C object src/CMakeFiles/valkey-server.dir/rio.c.o
[ 52%] Building C object src/CMakeFiles/valkey-server.dir/rand.c.o
[ 52%] Building C object src/CMakeFiles/valkey-server.dir/memtest.c.o
[ 53%] Building C object src/CMakeFiles/valkey-server.dir/syscheck.c.o
[ 53%] Building C object src/CMakeFiles/valkey-server.dir/crcspeed.c.o
[ 54%] Building C object src/CMakeFiles/valkey-server.dir/crccombine.c.o
[ 54%] Building C object src/CMakeFiles/valkey-server.dir/crc64.c.o
[ 55%] Building C object src/CMakeFiles/valkey-server.dir/bitops.c.o
[ 55%] Building C object src/CMakeFiles/valkey-server.dir/sentinel.c.o
[ 56%] Building C object src/CMakeFiles/valkey-server.dir/notify.c.o
[ 56%] Building C object src/CMakeFiles/valkey-server.dir/setproctitle.c.o
[ 57%] Building C object src/CMakeFiles/valkey-server.dir/blocked.c.o
[ 57%] Building C object src/CMakeFiles/valkey-server.dir/hyperloglog.c.o
[ 58%] Building C object src/CMakeFiles/valkey-server.dir/latency.c.o
[ 58%] Building C object src/CMakeFiles/valkey-server.dir/sparkline.c.o
[ 59%] Building C object src/CMakeFiles/valkey-server.dir/valkey-check-rdb.c.o
[ 59%] Building C object src/CMakeFiles/valkey-server.dir/valkey-check-aof.c.o
[ 60%] Building C object src/CMakeFiles/valkey-server.dir/geo.c.o
[ 60%] Building C object src/CMakeFiles/valkey-server.dir/lazyfree.c.o
[ 61%] Building C object src/CMakeFiles/valkey-server.dir/module.c.o
[ 61%] Building C object src/CMakeFiles/valkey-server.dir/evict.c.o
[ 62%] Building C object src/CMakeFiles/valkey-server.dir/expire.c.o
[ 62%] Building C object src/CMakeFiles/valkey-server.dir/geohash.c.o
[ 63%] Building C object src/CMakeFiles/valkey-server.dir/geohash_helper.c.o
[ 63%] Building C object src/CMakeFiles/valkey-server.dir/childinfo.c.o
[ 64%] Building C object src/CMakeFiles/valkey-server.dir/allocator_defrag.c.o
[ 64%] Building C object src/CMakeFiles/valkey-server.dir/defrag.c.o
[ 65%] Building C object src/CMakeFiles/valkey-server.dir/siphash.c.o
[ 65%] Building C object src/CMakeFiles/valkey-server.dir/rax.c.o
[ 66%] Building C object src/CMakeFiles/valkey-server.dir/t_stream.c.o
[ 66%] Building C object src/CMakeFiles/valkey-server.dir/listpack.c.o
[ 67%] Building C object src/CMakeFiles/valkey-server.dir/localtime.c.o
[ 67%] Building C object src/CMakeFiles/valkey-server.dir/lolwut.c.o
[ 68%] Building C object src/CMakeFiles/valkey-server.dir/lolwut5.c.o
[ 68%] Building C object src/CMakeFiles/valkey-server.dir/lolwut6.c.o
[ 69%] Building C object src/CMakeFiles/valkey-server.dir/acl.c.o
[ 69%] Building C object src/CMakeFiles/valkey-server.dir/tracking.c.o
[ 70%] Building C object src/CMakeFiles/valkey-server.dir/socket.c.o
[ 70%] Building C object src/CMakeFiles/valkey-server.dir/tls.c.o
[ 71%] Building C object src/CMakeFiles/valkey-server.dir/rdma.c.o
[ 71%] Building C object src/CMakeFiles/valkey-server.dir/sha256.c.o
[ 72%] Building C object src/CMakeFiles/valkey-server.dir/timeout.c.o
/Users/jbergstroem/wrk/oss/valkey/src/timeout.c:169:21: error: implicit conversion from 'long long' to 'long double' changes value from 9223372036854775807 to 9223372036854775808 [-Werror,-Wimplicit-const-int-float-conversion]
  169 |         if (ftval > LLONG_MAX) {
      |                   ~ ^~~~~~~~~
/opt/homebrew/Cellar/llvm/19.1.7_1/lib/clang/19/include/limits.h:109:20: note: expanded from macro 'LLONG_MAX'
  109 | #define LLONG_MAX  __LONG_LONG_MAX__
      |                    ^~~~~~~~~~~~~~~~~
<built-in>:63:27: note: expanded from macro '__LONG_LONG_MAX__'
   63 | #define __LONG_LONG_MAX__ 9223372036854775807LL
      |                           ^~~~~~~~~~~~~~~~~~~~~
1 error generated.
make[2]: *** [src/CMakeFiles/valkey-server.dir/timeout.c.o] Error 1
make[1]: *** [src/CMakeFiles/valkey-server.dir/all] Error 2
make: *** [all] Error 2
```

Building with `make` like this:

```bash
CC=/usr/local/bin/clang CXX=/usr/local/bin/clang++ make
...
    CC timeout.o
timeout.c:169:21: warning: implicit conversion from 'long long' to 'long double' changes value from 9223372036854775807 to 9223372036854775808 [-Wimplicit-const-int-float-conversion]
  169 |         if (ftval > LLONG_MAX) {
      |                   ~ ^~~~~~~~~
/opt/homebrew/Cellar/llvm/19.1.6/lib/clang/19/include/limits.h:109:20: note: expanded from macro 'LLONG_MAX'
  109 | #define LLONG_MAX  __LONG_LONG_MAX__
      |                    ^~~~~~~~~~~~~~~~~
<built-in>:64:27: note: expanded from macro '__LONG_LONG_MAX__'
   64 | #define __LONG_LONG_MAX__ 9223372036854775807LL
      |                           ^~~~~~~~~~~~~~~~~~~~~
1 warning generated.
```

where `/usr/local/bin/clang` & `/usr/local/bin/clang++` are symbolic links to `clang 19`, so it seems that `cmake` is more strict than the Makefile, I will look into this

--- a/src/timeout.c
+++ b/src/timeout.c
@@ -166,7 +166,7 @@ int getTimeoutFromObjectOrReply(client *c, robj *object, mstime_t *timeout, int 
             return C_ERR;
 
         ftval *= 1000.0; /* seconds => millisec */
-        if (ftval > LLONG_MAX) {
+        if (ftval > (long double)LLONG_MAX) {
             addReplyError(c, "timeout is out of range");
             return C_ERR;
         }

